src/engine/bank02.asm

```diff
@@ -426,117 +2184,154 @@ GetPointerToDeckCards: ; 9048 (2:5048)
 	pop af
 	ret
 
-Func_905a: ; 905a (2:505a)
+ResetCursorPosAndBlink: ; 905a (2:505a)
 	xor a
-	ld [wceaf], a
-	ld [wceb0], a
-	ld [wcea3], a
+	ld [wCursorDuelXPosition], a
+	ld [wCursorDuelYPosition], a
+	ld [wDuelCursorBlinkCounter], a
 	ret
 
-Func_9065: ; 9065 (2:5065)
+; handle player input in menu in Your Play Area
+; works out which cursor coordinate to go to
+; and sets carry flag if A or B are pressed
+; input
+; returns a =  $1 if A pressed
+; returns a = $ff if B pressed
+HandleDuelMenuInput_YourPlayArea: ; 9065 (2:5065)
 	xor a
 	ld [wcfe3], a
-	ld a, [wceaf]
+	ld a, [wCursorDuelXPosition]
 	ld d, a
-	ld a, [wceb0]
+	ld a, [wCursorDuelYPosition]
 	ld e, a
+
+; d = cursor x position
+; e = cursor y position
+
 	ldh a, [hDPadHeld]
 	or a
-	jr z, .asm_90a6
+	jr z, .no_pad
 	bit D_LEFT_F, a
-	jr nz, .asm_907e
+	jr nz, .horizontal
 	bit D_RIGHT_F, a
-	jr z, .asm_9084
-.asm_907e
+	jr z, .check_vertical
+
+; handle horizontal input
+.horizontal
 	ld a, d
-	xor $1
+	xor $1 ; flips x coordinate
 	ld d, a
-	jr .asm_9090
-.asm_9084
+	jr .okay
+.check_vertical
 	bit D_UP_F, a
-	jr nz, .asm_908c
+	jr nz, .vertical
 	bit D_DOWN_F, a
-	jr z, .asm_90a6
-.asm_908c
+	jr z, .no_pad
+
+; handle vertical input
+.vertical
 	ld a, e
-	xor $1
+	xor $01 ; flips y coordinate
 	ld e, a
-.asm_9090
-	ld a, $1
+
+.okay
+	ld a, $01
 	ld [wcfe3], a
 	push de
-	call Func_90d8
+	call DrawCursorEmpty_YourPlayArea
 	pop de
+
+;update x and y cursor positions
 	ld a, d
-	ld [wceaf], a
+	ld [wCursorDuelXPosition], a
 	ld a, e
-	ld [wceb0], a
+	ld [wCursorDuelYPosition], a
+	
+; reset cursor blink
 	xor a
-	ld [wcea3], a
-.asm_90a6
+	ld [wDuelCursorBlinkCounter], a
+.no_pad
 	ldh a, [hKeysPressed]
 	and A_BUTTON | B_BUTTON
-	jr z, .asm_90c1
+	jr z, .no_input
 	and A_BUTTON
-	jr nz, .asm_90b7
-	ld a, $ff
-	call Func_90fb
+	jr nz, .a_press
+	ld a, $ff ; cancel
+	call PlaySFXConfirmOrCancel
 	scf
 	ret
-.asm_90b7
+
+.a_press
 	call Func_90f7
-	ld a, $1
-	call Func_90fb
+	ld a, $01
+	call PlaySFXConfirmOrCancel
 	scf
 	ret
-.asm_90c1
+
+.no_input
 	ld a, [wcfe3]
 	or a
-	jr z, .asm_90ca
+	jr z, .check_blink
 	call PlaySFX
-.asm_90ca
-	ld hl, wcea3
+
+.check_blink
+	ld hl, wDuelCursorBlinkCounter
 	ld a, [hl]
 	inc [hl]
-	and $f
-	ret nz
-	ld a, $f
-	bit 4, [hl]
-	jr z, asm_90da
-Func_90d8: ; 90d8 (2:50d8)
-	ld a, $0
-asm_90da
+	and %00001111
+	ret nz  ; only update cursor if blink's lower nibble is 0
+
+	ld a, $0f ; cursor byte
+	bit 4, [hl] ; only draw cursor if blink counter's fourth bit is not set
+	jr z, DrawByteInCursor_YourPlayArea
+
+; draws in the cursor position
```

Comment:
I think you gave the trailing _YourPlayArea_ to ``DrawCursorEmpty_YourPlayArea`` and ``DrawByteInCursor_YourPlayArea`` following that of ``HandleDuelMenuInput_YourPlayArea``, so for the same reasons as above I would replace it in these two. I would maybe use ``EraseCheckMenuCursor`` and ``DrawCheckMenuCursor``.

Reply:
My reasoning for choosing "Byte" instead of cursor is that the function is more generic than that, since it draws whatever is in a. I don't know what the convention is for these kinds of things, whether to name it based on the function by itself or by how it is used consistently. Indeed, this function is only called (thus far) with the cursor in a.

Reply:
Ultimately I think it comes down to subtle details, as, to me, ``DrawByteInCursor`` and ``DrawCursor`` kind of say the same thing. Even if the function accept anything as the "byte", in practice, it's always going to be one of the symbol ID's corresponding to a cursor, or maybe a white (empty) tile. The reason it accepts any single byte is for convenience since there is perhaps more than one possible tile and it saves some space if the caller can choose it.

So having said that, I think I would choose DrawCursor over DrawByteInCursor on the basis that while both say the same thing, the former just sounds better to me. To me, it's like, say, if you had a function that prints the type icon of a card. It would probably accept any byte, but in practice only expect those that correspond to type constants or type icons. On the other hand, you may have an utility function in bank0 that loads an array of bytes at hl to a multi-purpose buffer in some address. For me, that would be the distinction between a specific function and a generic function.

Reply:
Sure, fair enough. It's my first time disassembling so learning these conventions is always good.